lib/resty/core/regex.lua

```diff
@@ -343,15 +343,15 @@ local function re_match_compile(regex, opts)
 end
 
 
-local function re_match_helper(subj, regex, opts, ctx, want_caps, res, nth)
+local function re_match_helper(subj, regex, opts, ctx, only_caps, res, nth, want_caps)
```

Comment:
I took the liberty of renaming in order to carry the meaning of "wanting *only* captures" vs "wanting captures *too*"